tests/appsec/contrib_appsec/test_fastapi.py

```diff
@@ -56,7 +72,18 @@ def patch_get(*args, **kwargs):
                 # httpx does not accept unicode headers and is now used in the TestClient
                 if "headers" in kwargs and FASTAPI_VERSION >= (0, 87, 0):
                     kwargs["headers"] = {k.encode(): v.encode() for k, v in kwargs["headers"].items()}
-                return initial_get(*args, **kwargs, allow_redirects=False)
+                if HTTPX_VERSION >= (0, 18, 0) and STARLETTE_VERSION >= (0, 21, 0):
+                    if "cookies" in kwargs:
+                        client.cookies = kwargs["cookies"]
+                        del kwargs["cookies"]
+                    else:
+                        client.cookies = {}
+                    if "data" in kwargs and not isinstance(kwargs["data"], dict):
+                        kwargs["content"] = kwargs["data"]
+                        del kwargs["data"]
+                if redirect_key not in kwargs:
+                    kwargs[redirect_key] = False
```

Comment:
Looks like we could extract this block of logic to another function (as it is used twice)

Reply:
true 👍